job/replay_worker_test.go

```diff
@@ -0,0 +1,160 @@
+
+			worker := job.NewReplayWorker(replaySpecRepoFac, scheduler)
+			err := worker.Process(ctx, replayRequest)
+			assert.NotNil(t, err)
+			assert.True(t, strings.Contains(err.Error(), errorMessage))
+		})
+		t.Run("should throw an error when updatestatus throws an error for failed request", func(t *testing.T) {
+			logger.Init(logger.ERROR)
+			ctx := context.Background()
+			replayRepository := new(mock.ReplayRepository)
+			defer replayRepository.AssertExpectations(t)
+			replayRepository.On("UpdateStatus", currUUID, models.ReplayStatusInProgress, models.ReplayMessage{}).Return(nil)
+			errMessage := "error while clearing dag runs for job job-name: scheduler clear error"
+			failedReplayMessage := models.ReplayMessage{
+				Type:    job.AirflowClearDagRunFailed,
+				Message: errMessage,
+			}
+			updateStatusErr := errors.New("error while updating status to failed")
+			replayRepository.On("UpdateStatus", currUUID, models.ReplayStatusFailed, failedReplayMessage).Return(updateStatusErr)
+
+			replaySpecRepoFac := new(mock.ReplaySpecRepoFactory)
+			defer replaySpecRepoFac.AssertExpectations(t)
+			replaySpecRepoFac.On("New", replayRequest.Job).Return(replayRepository)
+
+			scheduler := new(mock.MockScheduler)
+			defer scheduler.AssertExpectations(t)
+			errorMessage := "scheduler clear error"
+			scheduler.On("Clear", ctx, replayRequest.Project, "job-name", dagRunStartTime, dagRunEndTime).Return(errors.New(errorMessage))
+
+			worker := job.NewReplayWorker(replaySpecRepoFac, scheduler)
+			err := worker.Process(ctx, replayRequest)
+			assert.NotNil(t, err)
+			assert.True(t, strings.Contains(err.Error(), updateStatusErr.Error()))
+		})
+
+		t.Run("should throw an error when updatestatus throws an error for successful request", func(t *testing.T) {
+			logger.Init(logger.ERROR)
+			ctx := context.Background()
+			replayRepository := new(mock.ReplayRepository)
+			defer replayRepository.AssertExpectations(t)
+			replayRepository.On("UpdateStatus", currUUID, models.ReplayStatusInProgress, models.ReplayMessage{}).Return(nil)
+			updateSuccessStatusErr := errors.New("error while updating replay request")
+			replayRepository.On("UpdateStatus", currUUID, models.ReplayStatusSuccess, models.ReplayMessage{}).Return(updateSuccessStatusErr)
+
+			replaySpecRepoFac := new(mock.ReplaySpecRepoFactory)
+			defer replaySpecRepoFac.AssertExpectations(t)
+			replaySpecRepoFac.On("New", replayRequest.Job).Return(replayRepository)
+
+			scheduler := new(mock.MockScheduler)
+			defer scheduler.AssertExpectations(t)
+			scheduler.On("Clear", ctx, replayRequest.Project, "job-name", dagRunStartTime, dagRunEndTime).Return(nil)
+
+			worker := job.NewReplayWorker(replaySpecRepoFac, scheduler)
+			err := worker.Process(ctx, replayRequest)
+			assert.NotNil(t, err)
+			assert.True(t, strings.Contains(err.Error(), updateSuccessStatusErr.Error()))
```

Comment:
Have you tried using `assert.Error` does that not work here?

Reply:
I have updated other instances also where we were using assert.NotNil(t, err)